core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.spec.ts

```diff
@@ -48,25 +49,45 @@ class TestEngineRunner {
   optimizedSsrEngine: OptimizedSsrEngine;
   engineInstance: NgExpressEngineInstance;
 
-  constructor(options: SsrOptimizationOptions, renderTime?: number) {
+  constructor(
+    options: SsrOptimizationOptions,
+    renderTime?: number,
+    params?: { withError?: boolean }
+  ) {
     // mocked engine instance that will render test output in 100 milliseconds
     const engineInstanceMock = (
       filePath: string,
       _: any,
       callback: SsrCallbackFn
     ) => {
       setTimeout(() => {
-        callback(undefined, `${filePath}-${this.renderCount++}`);
+        const result = `${filePath}-${this.renderCount++}`;
+
+        if (params?.withError) {
+          const err = new Error(result);
+          callback(err, undefined);
+        } else {
+          callback(undefined, result);
+        }
       }, renderTime ?? defaultRenderTime);
     };
 
-    this.optimizedSsrEngine = new OptimizedSsrEngine(
-      engineInstanceMock,
-      options
-    );
+    this.optimizedSsrEngine = new OptimizedSsrEngine(engineInstanceMock, {
+      shouldCacheRenderingResult:
+        defaultSsrOptimizationOptions.shouldCacheRenderingResult,
```

Comment:
[QUESTION]
Do we really need to pass this option explicitly? Since it's a default, won't it just work without this line?

Reply:
Indeed, such option doesn't have to be pased explicitly. Looks like a leftover. Thanks!